pkg/provider/basic.go

```diff
@@ -165,6 +167,11 @@ func (bo *basicOperations) RefreshToken(ctx context.Context, t *Token, opts ...R
 		return nil, semerr.Map(err)
 	}
 
+	tokenExpiry := cast.ToInt64(o.ProviderOptions["token_expiry"])
+	if tok.Expiry.IsZero() && tokenExpiry > 0 {
```

Comment:
Perhaps this should look more like a `min(actual_token_expiry, requested_token_expiry)` operation? So users could use this as an override too if they want for some reason.